### ISSUE TITLE
Javascript errors in IE9; graph doesn't display

### ISSUE THREAD

**Issue by GoogleCodeExporter** [josejaner/flot#480]
```
In the release candidate of Internet Explorer 9, I cannot get flot to render a 
graph under any circumstances.  The system works fine in all other browsers, 
including earlier versions of IE.  I am using jQuery 1.5.1.  Using the DebugBar 
plugin, I was able to obtain the following error message:

----------------------------

Line: 554
Character: 17
Code: 0
Error Message: Unable to get value of the property 'init_': object is null or 
undefined
URL: http://server/js/flot/jquery.flot.js

----------------------------

The error occurs whenever the graph is plotted, not when the flot library is 
loaded.  Every graph I attempt to plot fails in IE9, but if necessary, I can 
grant access to one of the specific pages that causes this error if someone 
requests it.
```

Original issue reported on code.google.com by `shoemake...@gmail.com` on 12 Mar 2011 at 1:42

**Comment by GoogleCodeExporter** [comment]
```
Hi, thanks for the report. It was fixed a long time ago in the development 
branch, which is about to be released. I think I'll keep this open until I 
actually make the release.
```

Original comment by `olau%iol...@gtempaccount.com` on 14 Mar 2011 at 12:41

**Comment by GoogleCodeExporter** [comment]
```
Now that IE9 is RTW, how quickly can we get an official stable point release of 
flot that is not broken for IE9? We would like to encourage all our IE users to 
feel safe upgrading but we can't until flot supports IE9!
```

Original comment by `znbai...@gmail.com` on 16 Mar 2011 at 2:43

**Comment by GoogleCodeExporter** [comment]
```
I could also use this update as soon as possible.  I've already got users on my 
website with IE9 reporting the page not loading due to this graphing bug.  I 
have to disable graphing for now which is a big part of the website.
```

Original comment by `bryceray...@gmail.com` on 16 Mar 2011 at 3:35

**Comment by GoogleCodeExporter** [comment]
```
As an interim, just use the development branch?  It's not a huge change, mostly 
the move to plugins for some functionality.
```

Original comment by `ryl...@gmail.com` on 16 Mar 2011 at 4:01

**Comment by GoogleCodeExporter** [comment]
```
For others having this problem, I was able to work around it by changing the 
flot code to detect browser capabilities by perform feature detection instead 
of user-agent sniffing.

I did this by replacing $.browser.msie (jquery's flag for determining if the 
current browser is any version of IE) with an appropriate check for the window 
object the subsequent line is using.

Example:

if ($.browser.msie) // excanvas hack
        c = window.G_vmlCanvasManager.initElement(c);

gets changed to

if (window.G_vmlCanvasManager) // excanvas hack
        c = window.G_vmlCanvasManager.initElement(c);

That appears to have fixed our problems at least and may be a more comfortable 
solution for some rather than doing a larger upgrade to an unstable development 
branch version.
```

Original comment by `znbai...@gmail.com` on 16 Mar 2011 at 4:08

**Comment by GoogleCodeExporter** [comment]
```
You can also fix it by changing the HTML include for excanvas to read <!--[if 
lte IE 8]> instead of the old [if IE] thing. Actually, this is what you should 
do as there's no reason for IE 9 users to download the excanvas script.
```

Original comment by `olau%iol...@gtempaccount.com` on 16 Mar 2011 at 8:46

**Comment by GoogleCodeExporter** [comment]
```
@6 good point that IE9 users no longer need excanvas, though not including it 
won't actually fix the flot error because that will not stop the flot JS from 
attempting to access the non-existent window variable G_vmlCanvasManager, as 
you can see above. The correct fix is as mentioned above, to convert the 
user-agent sniffing method to a feature detection approach.
```

Original comment by `znbai...@gmail.com` on 16 Mar 2011 at 9:47

**Comment by GoogleCodeExporter** [comment]
```
Ah, yes, sorry about that.
```

Original comment by `olau%iol...@gtempaccount.com` on 17 Mar 2011 at 9:59

**Comment by GoogleCodeExporter** [comment]
```
Hi I have also the same issue :-(

I just installed the brand news official IE9 and my page just blown up !
It is working on IE6/7 and 8, Chrome (latest version), Firefox (3.6.14), 
Opera(11.01), Safari (iPhone iOS 4.2.1), Safari W7 (5.0.3), lunascape(6.4.2), 
seaMonkey (1.1.2).

Does NOT display on OpenVMS using FireFox/Mozilla based on SeaMonkey 1.1.2
Neither on W7 using IE9 !

The fact it does NOT display on IE9 prevents me to successfully pass the 
"acceptation" tests !

I can NOT go to the production environment !

Can you HELP me !

Thanks in advance.
Dominique
On OpenVMS browser based on SeaMonkey 1.1.2 it 
```

Original comment by `dominiqu...@gmail.com` on 18 Mar 2011 at 1:07

**Comment by GoogleCodeExporter** [comment]
```
There were two working suggestions already in the above comments? Try comment 
#4 or #5. Unless a major problem shows up, I'm going to release 0.7 unchanged 
from the current development version in a couple of days:

http://flot.googlecode.com/svn/trunk/
```

Original comment by `olau%iol...@gtempaccount.com` on 18 Mar 2011 at 2:54

**Comment by GoogleCodeExporter** [comment]
```
Hi,

I downloaded the jquery.flot.js and it .. works !
GREAT !

Do I have to copy ALL the files or just the jquery.flot.js ?

I have another operating system on which I'm running a browser based on 
seaMonkey 1.1.2, and it does not display the graph, as the javascript stops on 
error : 

Error: uncaught exception: [Exception... "Component returned failure code: 
0x80070057 (NS_ERROR_ILLEGAL_VALUE) [nsIDOMHTMLCanvasElement.getContext]"  
nsresult: "0x80070057 (NS_ERROR_ILLEGAL_VALUE)"  location: "JS frame :: 
http://l15s36.sabin.railb.be/tools/jquery_flot.js :: makeCanvas :: line 701"  
data: no]

Can you help me with this ?

That page is perfectly displayed on all the other browsers but on OpenVMS :-(

PS : The file nalme is correct : jquery_info.txt (undescore is correct, exists 
and used)

MANY THANKS in advance,
Best regards,
Dominique
```

Original comment by `dominiqu...@gmail.com` on 18 Mar 2011 at 5:15

**Comment by GoogleCodeExporter** [comment]
```
jquery.flot.js is probably fine. About the SeaMonkey thing - it seems HTML 
canvas support is not enabled on that platform. Can't help you with that, it's 
probably a browser limitation. Ask the SeaMonkey people about it.
```

Original comment by `olau%iol...@gtempaccount.com` on 21 Mar 2011 at 1:18

**Comment by GoogleCodeExporter** [comment]
```
Thanks for your reply ;-)
Any idea how to test the HTML canvas support capacity ?

```

Original comment by `dominiqu...@gmail.com` on 22 Mar 2011 at 8:35

**Comment by GoogleCodeExporter** [comment]
```
If you're just interested in a simple example to test with, there's a little 
self-contained one here:

http://devfiles.myopera.com/articles/418/canvasExample.html

If you're interested in testing at runtime whether there's canvas support, 
well, I haven't tried it myself, but Mark Pilgrim has a suggestion here:

http://diveintohtml5.org/detect.html#canvas
```

Original comment by `olau%iol...@gtempaccount.com` on 22 Mar 2011 at 10:31

**Comment by GoogleCodeExporter** [comment]
```
Thanks :-)
You are right ... no support of Canvas on my web browser running under OpenVMS 
:'(
```

Original comment by `dominiqu...@gmail.com` on 22 Mar 2011 at 11:20

**Comment by GoogleCodeExporter** [comment]
```
I've now released 0.7, so I'll close this.
```

Original comment by `olau%iol...@gtempaccount.com` on 22 Mar 2011 at 12:30
- Changed state: **Fixed**
